lib/url.c

```diff
@@ -3459,6 +3459,10 @@ static CURLcode create_conn(struct Curl_easy *data,
    * Process the "connect to" linked list of hostname/port mappings.
    * Do this after the remote port number has been fixed in the URL.
    *************************************************************/
+#ifndef CURL_DISABLE_ALTSVC
+  if(data->asi)
+    data->asi->used = TRUE;
```

Comment:
`data->asi` is the pointer to the "altsvc cache". It just means that curl has loaded or created such a cache in memory. it does not mean that any host matched. This condition is therefore incorrect.